pkg/controller/core/localqueue_controller.go

```diff
@@ -174,11 +196,75 @@ func (h *qWorkloadHandler) Generic(e event.GenericEvent, q workqueue.RateLimitin
 	q.AddAfter(req, constants.UpdatesBatchPeriod)
 }
 
+// qCQHandler signals the controller to reconcile the Queue associated
+// to the workload in the event.
+type qCQHandler struct {
+	client client.Client
+}
+
+func (h *qCQHandler) Create(e event.CreateEvent, wq workqueue.RateLimitingInterface) {
+	cq, ok := e.Object.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	h.addLocalQueueToWorkQueue(cq, wq)
+}
+
+func (h *qCQHandler) Update(e event.UpdateEvent, wq workqueue.RateLimitingInterface) {
+	newCq, ok := e.ObjectNew.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	oldCq, ok := e.ObjectOld.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	// Iff .status.conditions of the clusterQueue is updated,
+	// this handler sends all queues related to the clusterQueue to workqueue.
+	if equality.Semantic.DeepEqual(oldCq.Status.Conditions, newCq.Status.Conditions) {
+		return
+	}
+	h.addLocalQueueToWorkQueue(newCq, wq)
+}
+
+func (h *qCQHandler) Delete(e event.DeleteEvent, wq workqueue.RateLimitingInterface) {
+	cq, ok := e.Object.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	h.addLocalQueueToWorkQueue(cq, wq)
+}
+
+func (h *qCQHandler) Generic(event.GenericEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) addLocalQueueToWorkQueue(cq *kueue.ClusterQueue, wq workqueue.RateLimitingInterface) {
+	ctx := context.TODO()
+	log := ctrl.LoggerFrom(ctx).WithValues("clusterQueue", klog.KObj(cq))
+	ctx = ctrl.LoggerInto(ctx, log)
+	var queues kueue.LocalQueueList
+	err := h.client.List(ctx, &queues, client.MatchingFields{utilindexer.QueueClusterQueueKey: cq.Name})
+	if err != nil {
+		log.Error(err, "listing queues that match the clusterQueue")
+		return
+	}
+	for _, q := range queues.Items {
+		req := reconcile.Request{
+			NamespacedName: client.ObjectKeyFromObject(&q),
+		}
+		wq.AddAfter(req, constants.UpdatesBatchPeriod)
+	}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *LocalQueueReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	queueCQHandler := qCQHandler{
+		client: r.client,
+	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&kueue.LocalQueue{}).
 		Watches(&source.Channel{Source: r.wlUpdateCh}, &qWorkloadHandler{}).
+		Watches(&source.Kind{Type: &kueue.ClusterQueue{}}, &queueCQHandler).
```

Comment:
If you are using a Kind source, you should get the information about existence of CQ and status from the client.

If you use a Channel source, you should get all the information from the caches.

This is because a Channel source is based on cache updates, whereas a Kind source is based on client updates.

Reply:
@alculquicondor Let me make sure my understanding is correct.

In my understanding of your comment below, I should watch all events for clusterQueue in the localQueue controller instead of sending a generic event from the clusterQueue controller to the localQueue controller using the function `NotifyClusterQueueUpdate` function. 

> Wait, we don't need to add another notifier. The ClusterQueue notifier is enough. We can just check that the ClusterQueue also has the Active condition.

https://github.com/kubernetes-sigs/kueue/pull/623#discussion_r1135797721

https://github.com/kubernetes-sigs/kueue/blob/c2bd78b0ce7a3cf6880faaf01f079309ad7324fa/pkg/controller/core/localqueue_controller.go#L67-L80

Perhaps, were you meant just synchronizing active status conditions in localQueue with clusterQueue, not watching all events using kind Source?

This means "another notifier" is not `NotifyClusterQueueUpdate`, right?

Reply:
> In my understanding of your comment below, I should watch all events for clusterQueue in the localQueue controller instead of sending a generic event from the clusterQueue controller 

Yes, I think that's the simplest solution.

The notifiers are useful when you need to guarantee that the kueue cache is updated before updating the other object. If these cache updates are not reflected in the objects' status somehow, then we need to use notifiers.

But whether a ClusterQueue exists or has an active condition is visible in the status, so we can just use that.

Reply:
Thank you for the response.
My understanding looks like be correct.

Reply:
Done.